Sources/Sentry/SentryHttpTransport.m

```diff
@@ -139,6 +166,12 @@ - (void)sendCached:(NSURLRequest *)request withFilePath:(NSString *)filePath
             if (nil != response) {
                 [_self.fileManager removeFileAtPath:filePath];
                 [_self.rateLimits update:response];
+
+                [_self sendNextCachedEnvelope];
+            } else {
+                @synchronized(_self) {
+                    _self.isSending = NO;
```

Comment:
Do we need these private methods? It's super hard to follow the lock and this flag flipping.
I wrote 2 comments only to realize the private method call flipped the flag

Reply:
I'm going to remove it. 👍